src/components/Visualisers/Map.tsx

```diff
@@ -78,119 +65,152 @@ const Map: FC<{
-    sfSubgraph.useStreamsQuery(
-      {
-        chainId: network.chainId,
-        filter: {
-          sender: accountAddress,
-          currentFlowRate_not: "0",
-        },
-        pagination: {
-          take: Infinity,
-        },
+  const {
+    outgoingStreams,
+    outgoingNodeList,
+    outgoingEdgeList,
+    ...outgoingStreamsQuery
+  } = sfSubgraph.useStreamsQuery(
+    {
+      chainId: network.chainId,
+      filter: {
+        sender: accountAddress,
+        currentFlowRate_not: "0",
+      },
+      pagination: {
+        take: Infinity,
       },
-      {
-        selectFromResult: (queryResult) => ({
+    },
+    {
+      selectFromResult: (queryResult) => {
+        const outgoingStreams = queryResult.data?.items ?? [];
+
+        const outgoingNodeList = [
+          initialNodes[0],
+          ...outgoingStreams.map((stream, i: number) => {
+            let ethersFlowRate = ethers.utils.formatEther(
+              stream.currentFlowRate
+            );
+            //Convert to humanized monthly amount
+            let humanizedFlowRate = +ethersFlowRate * 3600 * 24 * 30;
+            //Create a node object with relevant data
+            let node: Node = {
+              id: `${stream.receiver}-${i}`,
+              position: { x: i * 150, y: 500 },
+              data: {
+                label: (
+                  <UserBlock network={network} address={stream.receiver} />
+                ),
+              },
+              flowRate: `${humanizedFlowRate.toFixed(2)}/Mo  ${
+                stream.tokenSymbol
+              }`,
+              sourcePosition: Position.Top,
+            };
+            return node;
+          }),
+        ];
+
+        const outgoingEdgeList: Edge[] = outgoingNodeList.map((node) => ({
+          id: `e${node.id}-${accountAddress}`,
+          label: node.flowRate,
+          source: accountAddress,
+          target: node.id,
+          animated: true,
+        }));
+
+        return {
           ...queryResult,
-          outgoingStreams: queryResult.data?.items ?? [],
-        }),
-      }
-    );
-
-  const { incomingStreams, ...incomingStreamsQuery } =
-    sfSubgraph.useStreamsQuery(
-      {
-        chainId: network.chainId,
-        filter: {
-          receiver: accountAddress,
-          currentFlowRate_not: "0",
-        },
-        pagination: {
-          take: Infinity,
-        },
+          outgoingStreams,
+          outgoingNodeList,
+          outgoingEdgeList,
+        };
       },
-      {
-        selectFromResult: (queryResult) => ({
+    }
+  );
+
+  const {
+    incomingStreams,
+    incomingNodeList,
+    incomingEdgeList,
+    ...incomingStreamsQuery
+  } = sfSubgraph.useStreamsQuery(
+    {
+      chainId: network.chainId,
+      filter: {
+        receiver: accountAddress,
+        currentFlowRate_not: "0",
+      },
+      pagination: {
+        take: Infinity,
+      },
+    },
+    {
+      selectFromResult: (queryResult) => {
```

Comment:
I moved the mappings to the RTK-Query's selector, thinking that it is highly optimized, not to be blocking UI etc, which it is, but for different reasons: https://github.com/reduxjs/reselect 

It can't break up the loop into chunks, and it's not using WebWorkers, so at some point the browser will need to run the big (~5k) iterations synchronously which could block the UI, so not too useful here...

To optimize on the event loop to avoid UI blocking, some async calls / setTimeouts would need to be used.

But using these selectors at least chunked the iterations into 2 chunks (incoming & outgoing), instead of one, and turns out the main culprit of browser freeze was the `animated: true` for the edges.